specs/electra/beacon-chain.md

```diff
@@ -386,23 +1084,35 @@ def process_deposit_receipt(state: BeaconState, deposit_receipt: DepositReceipt)
     )
 ```
 
-#### New `process_execution_layer_exit`
+#### New `process_execution_layer_withdrawal_request`
 
-*Note*: This function is new in Electra:EIP7002.
+*Note*: This function is new in Electra following EIP-7002 and EIP-7251.
 
 ```python
-def process_execution_layer_exit(state: BeaconState, execution_layer_exit: ExecutionLayerExit) -> None:
+def process_execution_layer_withdrawal_request(
+    state: BeaconState,
+    execution_layer_withdrawal_request: ExecutionLayerWithdrawalRequest
+) -> None:
+    amount = execution_layer_withdrawal_request.amount
+    is_full_exit_request = amount == FULL_EXIT_REQUEST_AMOUNT
+
+    # If partial withdrawal queue is full, only full exits are processed
+    if len(state.pending_partial_withdrawals) == PENDING_PARTIAL_WITHDRAWALS_LIMIT and not is_full_exit_request:
+        return
+
     validator_pubkeys = [v.pubkey for v in state.validators]
     # Verify pubkey exists
-    pubkey_to_exit = execution_layer_exit.validator_pubkey
-    if pubkey_to_exit not in validator_pubkeys:
+    request_pubkey = execution_layer_withdrawal_request.validator_pubkey
+    if request_pubkey not in validator_pubkeys:
         return
-    validator_index = ValidatorIndex(validator_pubkeys.index(pubkey_to_exit))
-    validator = state.validators[validator_index]
+    index = ValidatorIndex(validator_pubkeys.index(request_pubkey))
+    validator = state.validators[index]
 
     # Verify withdrawal credentials
     is_execution_address = validator.withdrawal_credentials[:1] == ETH1_ADDRESS_WITHDRAWAL_PREFIX
```

Comment:
This isn't quite right. It needs to check for `ETH1_ADDRESS_WITHDRAWAL_PREFIX` or `COMPOUNDING_WITHDRAWAL_PREFIX`. In EIP-7251, this is function is used:

https://github.com/ethereum/consensus-specs/blob/88f5dde0f973151e9df84ba6b15d1c67287df7ee/specs/_features/eip7251/beacon-chain.md?plain=1#L953

Which checks for 0x01 or 0x02:

https://github.com/ethereum/consensus-specs/blob/88f5dde0f973151e9df84ba6b15d1c67287df7ee/specs/_features/eip7251/beacon-chain.md?plain=1#L384-L388

Reply:
thanks for catching this! bad merge against the other stuff in Electra, fixing now

Reply:
https://github.com/ethereum/consensus-specs/pull/3668/commits/db2c3ab554f2a353b6634ccdaa7f182123e49aa3